java/j2ee.persistence/src/org/netbeans/modules/j2ee/persistence/unit/PUDataObject.java

```diff
@@ -174,21 +175,15 @@ public boolean parseDocument() {
         } else if (isModified()){//if it's isn't modified and persistenc eexits (parsed) no need to reparse
             try{
                 String oldVersion = persistence.getVersion();
-                java.io.InputStream is = getEditorSupport().getInputStream();
                 String version=Persistence.VERSION_1_0;
-                try {
+                try (InputStream is = getEditorSupport().getInputStream()) {
                     version=JPAParseUtils.getVersion(is);
                 } catch (SAXException ex) {
                     LOG.log(Level.INFO, null, ex);//persistence.xml may be corrupted, but no need to show exception dialog
                 }
-                finally
-                {
-                    is.close();
-                }
-                is = getEditorSupport().getInputStream();
                 Persistence newPersistence;
                 Persistence cleanPersistence;
-                try {
+                try (java.io.InputStream is = getEditorSupport().getInputStream()) {
```

Comment:
This should change the same was a ~20 lines above.

Reply:
Done